html/getResourceTypeList.py

```diff
@@ -49,11 +49,15 @@
 else:
 	criteriaStr = ''
 
-if planetID.isdigit() and int(planetID) > 0:
-	criteriaStr = criteriaStr + ' AND (specificPlanet = 0 OR specificPlanet = %(planetID)s)'
+if galaxy.isdigit() and int(galaxy) > 0:
+	if planetID.isdigit() and int(planetID) > 0:
+		criteriaStr = criteriaStr + ' AND (COALESCE(trto.specificPlanet, tResourceType.specificPlanet) = 0 OR COALESCE(trto.specificPlanet, tResourceType.specificPlanet) = %(planetID)s)'
+	else:
+		criteriaStr = criteriaStr + ' AND (COALESCE(trto.specificPlanet, tResourceType.specificPlanet) = 0 OR COALESCE(trto.specificPlanet, tResourceType.specificPlanet) IN (SELECT DISTINCT tPlanet.planetID FROM tPlanet, tGalaxyPlanet WHERE (tPlanet.planetID < 11) OR (tPlanet.planetID = tGalaxyPlanet.planetID AND tGalaxyPlanet.galaxyID = %(galaxy)s)))'
 else:
-	if galaxy.isdigit() and int(galaxy) > 0:
-		criteriaStr = criteriaStr + ' AND (specificPlanet = 0 OR specificPlanet IN (SELECT DISTINCT tPlanet.planetID FROM tPlanet, tGalaxyPlanet WHERE (tPlanet.planetID < 11) OR (tPlanet.planetID = tGalaxyPlanet.planetID AND tGalaxyPlanet.galaxyID = %(galaxy)s)))'
+	if planetID.isdigit() and int(planetID) > 0:
+		criteriaStr = criteriaStr + ' AND (tResourceType.specificPlanet = 0 OR tResourceType.specificPlanet = %(planetID)s)'
+
```

Comment:
Previously, if a planet ID was provided, no consideration was made about the galaxy criteria. This change accounts for the following cases:

- `planetID` without a `galaxyID`
- `planetID` with a `galaxyID`
- `galaxyID` without a `planetID`